### DISCUSSION TITLE
Use process.env at run time

### DISCUSSION THREAD

**Discussion by peabnuts123** [parcel-bundler/parcel#6673] [Q&A] [answered]
Hi, I'm trying to compile my node.js application with parcel; however, it is using `process.env` to read some environment variables at run time. This is critical as the config is passed in from the environment the code runs in (AWS Lambda) to tell it how to behave. Through this, I can deploy the same artifact through different environments.

I can't seem to get `process.env` to end up in my output at all. I am using `parcel@2.0.0-beta.2`. As per [this link](https://v2.parceljs.org/features/node-emulation/#%F0%9F%8C%B3-environment-variables) it suggests that basically `process.env` is, by default, intended to be statically compiled to some constants (a common use-case), but you can disable this behaviour with the following config in `package.json`:
```json
  "@parcel/transformer-js": {
    "inlineEnvironment": false
  }
```

I've added this, but instead of just leaving `process.env` in the source, it seems to just replace everything with `undefined`. This doesn't seem very useful, or correct. What am I missing here? How can I access `process.env` at runtime?

In terms of reproduction I am just using a simple script:
```javascript
const PROJECT_ID = process.env.PROJECT_ID;

console.log("PROJECT_ID: " + PROJECT_ID);
```

and getting in my output:
```javascript
!function(){"use strict";console.log("PROJECT_ID: "+undefined)}();
```

**Comment by mischnic** [comment, marked as answer]
You want Parcel to treat this as a build for Node, and not the browser. E.g. by adding `engines: { node: ">=12" }` to your package.json.

**Reply by peabnuts123** [reply]
GitHub has a bug atm so I can't mark this as accepted but yes this does seem to have changed the behaviour of parcel to produce the desired result.
